src/main/java/com/artipie/asto/Storage.java

```diff
@@ -99,6 +99,21 @@
      */
     CompletableFuture<Void> delete(Key key);
 
+    /**
+     * Removes all items with key prefix.
+     *
+     * @param prefix Key prefix.
+     * @return Completion or error signal.
+     */
+    default CompletableFuture<Void> deleteAll(final Key prefix) {
+        return this.list(prefix).thenCompose(
+            list -> CompletableFuture.allOf(
+                list.stream().map(this::delete)
+                    .toArray(CompletableFuture[]::new)
+            )
+        );
```

Comment:
I'm not sure if it's really necessary, but I'd suggest adding a check that key exists before remove it

Reply:
@olenagerasimova I think It's not necessary since if the key doesn't exist, `list` will not return it.